roles/network_plugin/calico/templates/calico-node.yml.j2

```diff
@@ -428,7 +432,7 @@ spec:
           hostPath:
             path: /run/xtables.lock
             type: FileOrCreate
-{% if calico_datastore == "kdd" %}
+{% if calico_datastore == "kdd" and not calico_ipam_host_local %}
```

Comment:
This looks wrong, look at the comment: _It's used when upgrading from host-local to calico-ipam_, we need this only when ipam is host-local. and you can see https://github.com/projectcalico/calico/blob/4efd1bfd914b0c59086531c8c5a5ac5b593c18b1/charts/calico/templates/calico-node.yaml#L641

Please let me know if I'm wrong.

Reply:
Yeah, this was also not obvious for me. If the user want to use host-local as IPAM, then migrating the host-local state to calico-ipam is wrong, because host-local should be used.

The manifest in the Helm Chart allows a not working setup, because it doesn't support calico as network backend with the host-local IPAM, see https://github.com/projectcalico/calico/pull/8642

In my setup, I use calico only for policy enforcing, i.e. `calico_backend: none` and `calico_ipam_host_local: true`. It is wrong, in these setup to migrate to the calico-ipam by removing the running state of host_local IPAM.
https://github.com/projectcalico/calico/blob/4efd1bfd914b0c59086531c8c5a5ac5b593c18b1/manifests/calico-policy-only.yaml#L4781

Reply:
sorry for the delay. you are right! We need mount this only when the ipam is calico-ipam.